### PR TITLE
Only run Bk on all python versions on master/release branches

### PR DESCRIPTION
**Pull request by gibsondan**
Summary:
Adding 3.9 (and soon 3.10) is good, but the sheer number of BK steps running on every PR is getting pretty overwhelming and hard to parse, and actual version differences are quite rare in my experience (to the point where I'd personally be fine waiting to see them in master). There's probably a cost control benefit here as well, but that wasn't the main reason I was inspired to send this out.

None of this is held strongly.

## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by prha — COMMENTED** [review]
Based on the percentages in here: https://threads.com/34417000494, what do you think about including both 3.8 and 3.9?  That's 75% of our logged userbase?

**Comment by gibsondan** [comment]
seems like the main thing we'd want this for is a backstop to catch somebody accidentally using a new feature introduced in https://docs.python.org/3/whatsnew/3.9.html - i'd be fine with that not showing up until master since it seems super rare to me (maybe until it happens at last once?)

**Comment by prha** [comment]
Yeah, that's true.  Okay, I'm 100% on board with this.

**Review by prha — APPROVED** [review]

File: .buildkite/dagster-buildkite/dagster_buildkite/utils.py [hunk 1]
```diff
@@ -94,3 +96,13 @@ def connect_sibling_docker_container(network_name, container_name, env_variable)
 
 def is_release_branch(branch_name: str):
     return branch_name.startswith("release-")
+
+
+def get_python_versions():
```
Comment (alangenfeld) [inline .buildkite/dagster-buildkite/dagster_buildkite/utils.py:101]: I think we should name this to make it clear its behavior deviates based on branch
